crates/solver/src/settlement_post_processing/optimize_score.rs

```diff
@@ -0,0 +1,56 @@
+use {
+    super::SettlementSimulating,
+    crate::{
+        objective_value::Inputs,
+        settlement::Settlement,
+        solver::score_computation::ScoreCalculator,
+    },
+    ethcontract::Address,
+    gas_estimation::GasPrice1559,
+    num::{BigRational, FromPrimitive},
+    shared::{external_prices::ExternalPrices, http_solver::model::Score},
+};
+
+pub async fn compute_score(
```

Comment:
From the perspective of this function it feels cleaner to take Settlement by reference and return the score. Up to you.

Reply:
Yes, I tried to fit the function into the context of the place where it is used, but it does not make it really reusable. Fixed.